src/dsp_tools/fast_xmlupload/process_files.py

```diff
@@ -102,7 +103,28 @@ def _process_files_in_parallel(
 
     orig_filepath_2_uuid: list[tuple[Path, Optional[Path]]] = []
     for processed in as_completed(processing_jobs):
-        orig_filepath_2_uuid.append(processed.result())
+        try:
+            orig_filepath_2_uuid.append(processed.result())
+        except docker.errors.APIError:
+            print(f"{datetime.now()}: ERROR: A Docker exception occurred. Restart SIPI and resume processing...")
+            logger.error("A Docker exception occurred. Restart SIPI and resume processing...", exc_info=True)
+            # cancel all running jobs
+            for job in processing_jobs:
+                job.cancel()
+            # restart sipi container
+            _stop_and_remove_sipi_container()
+            _start_sipi_container_and_mount_volumes(input_dir, output_dir)
+            global sipi_container
+            sipi_container = _get_sipi_container()
```

Comment:
perfect candidate to extract into a method `restart_sipi_container()` - then you don't even need the comment

Reply:
I refactored these methods, because they were a bit spurious anyway.